### PR TITLE
refactor: WIP RuleExtensions

### PR DESCRIPTION
**Pull request by Balcan**
WIP
Refactor to extract the code that translate SDK objects to Rule Engine objects.
WIP

### CODE REVIEW

**Comment by ferdyrod** [comment]
Just checkout the branch and the dashboard is not showing!

**Comment by JaimeToca** [comment]
is there any jira issue for this? can you link it here pls
Thanksss

File: app/src/main/java/org/dhis2/Bindings/RuleExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+            data()
+        )
+        ProgramRuleActionType.DISPLAYKEYVALUEPAIR -> RuleActionDisplayKeyValuePair.createForIndicators(
+            content(),
+            data()
+        )
+        ProgramRuleActionType.HIDESECTION -> programStageSection()?.let {
+            RuleActionHideSection.create(it.uid())
+        } ?: RuleActionUnsupported.create(
+            "HIDE SECTION RULE IS MISSING PROGRAM STAGE SECTION",
```
[the remark this answers is not in the document]
Comment (Balcan) [reply app/src/main/java/org/dhis2/Bindings/RuleExtensions.kt]: Copycat :P

**Comment by Balcan** [comment]
[related jira issue](https://jira.dhis2.org/browse/ANDROAPP-2461)

**Pull request merged**